### ISSUE TITLE
[GCE] API should be usable asynchronously

### ISSUE THREAD

**Issue by jschwartzentruber** [MozillaSecurity/laniakea#41]
Operations like `create()` and `terminate()` block execution until completed. There should be a way of polling for state changes as we do with EC2. Not sure if there is a precedent for this in libcloud or if we would have to abandon libcloud.

**Comment by posidron** [comment]
https://libcloud.apache.org/blog/2017/04/09/async-rfc.html

**Comment by jschwartzentruber** [comment]
Yeah, that's in the [API documentation](https://libcloud.readthedocs.io/en/latest/other/using-libcloud-in-multithreaded-and-async-environments.html) too, but that's for using py3 asyncio or gevent, etc. What 

I'm talking about is being able to hook provider polling so we can poll from another process using tokens or similar (eg. EC2 request ID). It looks like libcloud's async support works this way internally, but it isn't exposed directly.

I worked around this for termination in #42, and creation is fast in GCE, so it's less of an issue.
